### PR TITLE
Browse search prefs

### PR DESCRIPTION
**Pull request by sem42198**
## Description

A refactor of how search views in the staff interface which resolves a range of tickets related to consistency and which columns should show up in which places. Additionally, allows for more flexibility for setting which columns should show up for a given record type and how search results should be sorted by default (either in preferences or in a config file). Some new options for columns are included also such as dates and extents.

## Related JIRA Ticket or GitHub Issue

[ANW-681](https://archivesspace.atlassian.net/browse/ANW-681)
[ANW-685](https://archivesspace.atlassian.net/browse/ANW-685)
[ANW-929](https://archivesspace.atlassian.net/browse/ANW-929)
[ANW-540](https://archivesspace.atlassian.net/browse/ANW-540)
[ANW-216](https://archivesspace.atlassian.net/browse/ANW-216)
[ANW-215](https://archivesspace.atlassian.net/browse/ANW-215)
[ANW-249](https://archivesspace.atlassian.net/browse/ANW-249)

## How Has This Been Tested?

Ran all manual tests and tested views, csvs, etc. manually. Test server is available [here](http://searchprefs.lyrtech.org/staff/).

## Screenshots (if appropriate):
Screenshot from 2020-05-19 09-46-50

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have authority to submit this code.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: solr/schema.xml [hunk 1]
```diff
@@ -75,6 +75,10 @@
    <field name="notes" type="text_general" indexed="true" stored="true" multiValued="false"/>
    <field name="years" type="int" indexed="true" stored="false" multiValued="true"/>
    <field name="year_sort" type="sort_string" indexed="true" stored="false" multiValued="false"/>
+   <field name="dates" type="text_general" indexed="true" stored="true" multiValued="true"/>
+   <field name="extents" type="text_general" indexed="true" stored="true" multiValued="true"/>
+   <field name="context" type="text_general" indexed="true" stored="true" multiValued="true"/>
+   <field name="context_uri" type="string" indexed="true" stored="true" multiValued="true"/>
```
Comment (lorawoodford) [inline solr/schema.xml:81]: @sem42198 These should be removed as well post commit 4d99a35c398cfe871a6d34d4c8b9177ded4d6efa

**Comment by lorawoodford** [comment]
Hi @sem42198 -

I'm noticing that commas (possibly more?) are being escaped in the CSV export (though they are not being escaped in Solr).  This is most obvious if you've added a date column:

image

Also, while in the export csv output of an extents column (see `2 --- Leaves` in the above example) translations are being found for non-standard extent types, the UI is still complaining of translations missing: 

image

**Comment by sem42198** [comment]
@lorawoodford I'm not quite sure why the escaping of characters in the csv is happening but I'm not sure how much control we have over that. It is a multi-valued field in solr and it seems solr separated the values with an escaped comma (presumably because a non-escaped comma would put the data in a new cell). Ideally it would use a separator that could be rendered properly in excel. As for the translations missing, I added a fallback in my latest commit which should solve that problem.

File: solr/schema.xml (hunk 1 above)
Comment (sem42198) [reply solr/schema.xml]: Good catch. I've removed those fields

**Comment by mark-cooper** [comment]
Final (squashed) PR: https://github.com/archivesspace/archivesspace/pull/1913

**Pull request closed without merging**